elastalert/elastalert.py

```diff
@@ -481,6 +487,9 @@ def run_rule(self, rule, endtime, starttime=None):
             # Add it as an aggregated match
             self.add_aggregated_alert(match, rule)
 
+        # Mark this endtime for next run's start
+        rule['previous_endtime'] = endtime
```

Comment:
I'm not enthused by the choice of variable names here. `minimum_starttime`, `starttime`, `previous_endtime`, `endtime`, and `buffer_time` mix styles and don't state their intent very well at all. I feel that concern is orthogonal to the change you want to make here, however.

In the future, I recommend we clean up all of this intermixing of responsibility, first by better stating intent, and later by separating these logic into well-defined classes and interfacing.